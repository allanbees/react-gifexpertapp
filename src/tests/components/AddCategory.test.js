import { shallow } from "enzyme";
import { AddCategory } from "../../components/AddCategory";


describe('Tests of <AddCategory />', () => { 

    const setCategories = () => {};
    const wrapper = shallow(<AddCategory setCategories = { setCategories } />);

    test('Must render correctly', () => {

        expect( wrapper ).toMatchSnapshot();

    });

    test('Must change the text box', () => {

        const input = wrapper.find('input');
        const value = 'Hello world';

        // the second argument is the event we want to happen, in this case we need to send a string
        input.simulate('change', { target: { value }});
        expect( wrapper.find('p').text().trim() ).toBe( value ); 

    });



})