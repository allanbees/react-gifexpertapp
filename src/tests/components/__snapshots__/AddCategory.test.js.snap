// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests of <AddCategory /> Must render correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <p>
     
     
  </p>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
